vowpalwabbit/parser/flatbuffer/parse_flat_example.cc

```diff
@@ -0,0 +1,183 @@
+#include "generated/example_generated.h"
+
+namespace VW {
+namespace parsers {
+namespace flatbuffer {
+
+void read_flatbuffer(vw* all, char* line, size_t len, v_array<example*>& examples)
+{
+  // Do something
+}
+
+int flatbuffer_to_examples(vw* all, v_array<example*>& examples)
+{
+  return (int) all->flat_converter->parse_examples(all, examples); // Get rid of this
+}
+
+parser::parser(std::string filename) : 
+_filename(filename),
+_flatbuffer_pointer(nullptr),
+_example_index(0),
+_c_hash(0)
+{}
+
+parser::parser(uint8_t *buffer_pointer) : 
+_filename("empty"),
+_flatbuffer_pointer(buffer_pointer),
+_example_index(0),
+_c_hash(0)
+{
+  _data = VW::parsers::flatbuffer::GetExampleCollection(_flatbuffer_pointer); //Shift as a separate call
+}
+
+void parser::init()
+{
+  std::ifstream infile;
+  infile.open(_filename, std::ios::binary | std::ios::in);
+  if (!infile.good()) THROW_EX(VW::vw_argument_invalid_value_exception, "Flatbuffer does not exist");
+
+  infile.seekg(0,std::ios::end);
+  int length = infile.tellg();
+  infile.seekg(0,std::ios::beg);
+  std::unique_ptr<char> buffer_pointer(new char[length]);
+  infile.read(buffer_pointer.get(), length);
+  _flatbuffer_pointer = reinterpret_cast<u_int8_t*>(buffer_pointer.get());
+  infile.close();
+
+  _data = VW::parsers::flatbuffer::GetExampleCollection(_flatbuffer_pointer);
+}
+
+const VW::parsers::flatbuffer::ExampleCollection* parser::data() {return _data;}
+
+bool parser::parse_examples(vw* all, v_array<example*>& examples)
+{
+  if (!flatbuffers::IsFieldPresent(_data, ExampleCollection::VT_EXAMPLES))
+  {
+    THROW("No examples to parse");
+  }
+  if (_example_index == _data->examples()->size())
+    return false;
+  parse_example(all, examples[0], _data->examples()->Get(_example_index));
+  _example_index++;
+  return true;
+}
+
+void parser::parse_example(vw* all, example* ae, const Example* eg)
+{
+  all->p->lp.default_label(&ae->l);
+  parse_flat_label(all->sd, ae, eg);
+
+  if (flatbuffers::IsFieldPresent(eg, Example::VT_TAG)){
+    ae->tag = v_init<char>();
+    VW::string_view tag(eg->tag()->c_str());
+    push_many(ae->tag, tag.begin(), tag.size());
+  }
+
+  for (int i=0; i < eg->namespaces()->size(); i++){
+    parse_namespaces(all, ae, eg->namespaces()->Get(i));
+  }
+}
+
+void parser::parse_namespaces(vw* all, example* ae, const Namespace* ns)
+{
+  uint8_t temp_index;
+  if (flatbuffers::IsFieldPresent(ns, Namespace::VT_NAME)){
+    temp_index = (uint8_t)ns->name()->c_str()[0];
+    ae->indices.push_back(temp_index);
+    _c_hash = all->p->hasher(ns->name()->c_str(), ns->name()->Length(), all->hash_seed);
+  }
+  else {
+    ae->indices.push_back(ns->hash());
+    temp_index = ns->hash();
+  }
+
+  features& fs = ae->feature_space[temp_index];
+
+  for (int j=0; j<ns->features()->size(); j++){
+    parse_features(all, ae, fs, ns->features()->Get(j));
+  }
+}
+
+void parser::parse_features(vw* all, example* ae, features& fs, const Feature* feature)
```

Comment:
It looks like `example* ae` is not used here so it could be removed from the function's parameters